ci/test/03_test_script.sh

```diff
@@ -144,21 +144,74 @@ if [ "$RUN_CHECK_DEPS" = "true" ]; then
   "${BASE_ROOT_DIR}/contrib/devtools/check-deps.sh" .
 fi
 
+function get_interfaces()
+{
+  ifconfig | awk -F ':| ' '/^[^[:space:]]/ { if (!match($1, /^lo/)) { print $1 } }'
+}
+
+function tcpdump_file_for_interface()
+{
+  echo "/tmp/tcpdump_$1"
+}
+
+function traffic_monitor_begin()
+{
+  for ifname in $(get_interfaces) ; do
+    tcpdump -nU -i "$ifname" -w "$(tcpdump_file_for_interface "$ifname")" &
+  done
+}
+
+function traffic_monitor_end()
+{
+  # Stop all tcpdump instances (we want the word splitting if "jobs -p" returns more than one PID).
+  # shellcheck disable=SC2046
+  while kill -SIGTERM $(jobs -p) ; do
+    sleep 1
+  done
+
+  for ifname in $(get_interfaces) ; do
+    f=$(tcpdump_file_for_interface "$ifname")
+    if [ ! -e "$f" ] && [ "$FILE_ENV" != "./ci/test/00_setup_env_native_asan.sh" ] ; then
+      # In some CI environments this script is not running as root and so the
+      # tcpdump errors and does not create $f. Skip silently those, but we
+      # need at least one where tcpdump can run and this is the ASAN one. So
+      # treat the absence of $f as an error only on the ASAN task.
+      continue
+    fi
+    # We are running as root and those files are created with owner:group =
+    # tcpdump:tcpdump and then `tcpdump -r` refuses to read them with an error
+    # "permission denied" if they are not owned by root:root.
+    chown root:root "$f"
+    if [ -n "$(tcpdump -n -r "$f" --direction=out tcp or udp)" ] ; then
```

Comment:
Ignoring the return status of `tcpdump` here is deliberate because some VMs, the ones that couldn't stop `tcpdump` failed to read the unflushed file with an error like "tcpdump: truncated dump file; tried to read 4 file header bytes, only got 0". Now they don't due to the added `-U`, hmm...

Reply:
Ok, I changed this to fail if `tcpdump` gives error when it reads the file.

Reply:
I assume `ifconfig` without any arguments will never fail.

Do you think it would be better to change it to:

```diff
 function get_interfaces()
 {
+  set -o pipefail
   ifconfig | awk -F ':| ' '/^[^[:space:]]/ { if (!match($1, /^lo/)) { print $1 } }'
+  set +o pipefail
 }
```

or that is just clutter?

Reply:
Enabled `pipefail` as in the above diff. Thanks!